engine/lib/widgets.php

```diff
@@ -110,14 +122,22 @@ function elgg_is_widget_type($handler) {
  *
  * The widget types are \stdClass objects.
  *
- * @param string $context The widget context or empty string for current context
- * @param bool   $exact   Only return widgets registered for this context (false)
+ * @param array|string $context An associative array of options or the widget context
+ * @param bool         $exact   Only return widgets registered for this context (false)
  *
  * @return array
  * @since 1.8.0
  */
 function elgg_get_widget_types($context = "", $exact = false) {
-	return _elgg_services()->widgets->getTypes($context, $exact);
+	if (is_array($context)) {
+		$params = $context;
+	} else {
+		$params = [
+			'context' => $context,
+			'exact' => $exact,
```

Comment:
set `'container' => null` for readability? Maybe even `$default` array and then merge?